atlasdb-ete-tests/src/test/java/com/palantir/atlasdb/ete/MultiCassandraStartupOrderingEteTest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.ete;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import com.palantir.atlasdb.todo.ImmutableTodo;
+import com.palantir.atlasdb.todo.Todo;
+import com.palantir.atlasdb.todo.TodoResource;
+
+@RunWith(Parameterized.class)
+public class MultiCassandraStartupOrderingEteTest {
+    private final List<String> cassandraNodesToStartOrStop;
+
+    public MultiCassandraStartupOrderingEteTest(List<String> cassandraNodesToStartorStop) {
+        this.cassandraNodesToStartOrStop = cassandraNodesToStartorStop;
+    }
+
+    @Parameterized.Parameters
+    public static Collection<Object[]> testCases() {
+        Collection<Object[]> params = Lists.newArrayList();
+
+        params.add(new Object[] {ImmutableList.of("cassandra1")});
+        params.add(new Object[] {ImmutableList.of("cassandra1", "cassandra2")});
+        params.add(new Object[] {ImmutableList.of("cassandra1", "cassandra2", "cassandra3")});
+        return params;
+    }
+
+    @Test
+    public void shouldBeAbleToStartUpIfCassandraIsDownAndRunTransactionsWhenCassandraIsUp()
+            throws IOException, InterruptedException {
+        EteSetup.runCliCommand("service/bin/init.sh stop");
+        runOnCassandraNodes(MultiCassandraTestSuite::killCassandraContainer);
+        EteSetup.runCliCommand("service/bin/init.sh start");
```

Comment:
i might just be unfamiliar with the testing setup, but i don't know what the `init.sh` is doing when reading this. consider refactoring it into a method whose name describes what the command does.